### DISCUSSION TITLE
CBT 입점 API 문의

### DISCUSSION THREAD

**Discussion by YoungDaeJun** [commerce-api-naver/commerce-solutions#23] [묻고 답하기]
안녕하세요!
키위스냅입니다.

네이버 커머스API센터 CBT 참여하기로 하고 프로세스 검토중에 몇가지 문의 할 부분이 있습니다.

1. 애플리케이션 ID , 애플리케이션 시크릿 : 저희는 기존에 발급받은 ID, Secret 값으로 현재 상품등록/수정 등 연동을 하고 있습니다.
    커머스API 센터에 이번에 등록한 솔루션 애플리케이션 값하고 다른데...변경을 해서 호출을 해야 하는지요?
    변경을 해야 한다면 변경을해서 호출시 기존 연동한 상품들도 동일하게 수정이나 조회 연동이 가능한지 확인 부탁드립니다.
2. JWE 검증 사용
    저희는 현재 셀러분이 판매자센터에서 api 판매자ID 를 복사 후 키위스냅에 입력하여 사용중에 있습니다.
    이부분이 JWE 검증 관련 관련이 있는거 같은데.. 개발을 해야 하는지와...개발을 해야 한다면 어떻게 활용하면 되는지도 확인 부탁드립니다.
    바로사용? 심사후 승인? 이해가 되지 않습니다.
3. CBT 참여 필수 조건인 비즈월렛
    EVENT HOOK 등록 후 구독, 업그레이드 등 이벤트가 발생하는 것으로 알고 있는데..관련 문서가 있을까요?
    https://github.com/commerce-api-naver/solutions-early/wiki   404로 접근이 안되어서요..

확인 부탁드립니다.

**Comment by commerce-api-naver** [comment]
안녕하세요. 커머스API 이민석입니다.

먼저 '키위스냅'은 현재 API대행사 방식으로 서비스를 제공중으로 이에 맞춰 답변드립니다.

## 첫 번째 문의에 답변
솔루션 입점 시 판매자의 연동 방식에 따라 올바른 애플리케이션 인증 정보를 활용해주셔야 합니다.
애플리케이션 인증 정보가 변경되더라도 상품/주문 연동과 관련한 권한상의 변동이 있는 것은 아니나, 내부적으로 두 가지의 연동 방식이 다르기에 판매자 별 연동 방식에 따라 각각 별개 취급됩니다.
(커머스솔루션 구독-API대행사 설정 상호간의 호환 불가)
- 커머스솔루션 구독(사용)을 하신 사용자 분들은 '솔루션 애플리케이션' 정보로 호출
- API대행사로 설정하신 사용자 분들은 '대행사 애플리케이션' 정보로 호출

## 두 번째 문의에 답변
JWE 방식은 '심사 후 승인'형 솔루션에서만 제공합니다. (관련 내용 [참고](https://github.com/commerce-api-naver/commerce-solutions/wiki/%5B%EA%B0%80%EC%9D%B4%EB%93%9C%5D-%EC%BB%A4%EB%A8%B8%EC%8A%A4%EC%86%94%EB%A3%A8%EC%85%98-%EC%82%AC%EC%9A%A9%EC%9E%90-%EC%8B%9D%EB%B3%84%EC%9E%90-%EC%97%B0%EB%8F%99-%EB%B0%8F-%EC%82%AC%EC%9A%A9-%ED%95%B4%EC%A7%80-%ED%94%84%EB%A1%9C%EC%84%B8%EC%8A%A4#%EC%86%94%EB%A3%A8%EC%85%98-%EC%9C%A0%ED%98%95%EB%B3%84-%EC%82%AC%EC%9A%A9%EC%9E%90-%EC%8B%9D%EB%B3%84-%EB%B0%8F-api-%ED%98%B8%EC%B6%9C-%EA%B6%8C%ED%95%9C-%ED%9A%8D%EB%93%9D-%EB%B0%A9%EB%B2%95))
커머스솔루션 연동 유형 2가지 - [바로 사용형], [심사 후 승인형] 중 1개를 선택하실 때 심사 후 승인형을 사용하실 예정이시라면 JWE는 필수 구현 사항입니다.
JWE 연동과 관련하여는 아래 가이드 문서를 참고해주시기 바랍니다.
- [판매자 커머스ID 인증](https://github.com/commerce-api-naver/commerce-solutions/wiki/%5B%EA%B0%80%EC%9D%B4%EB%93%9C%5D-%EC%BB%A4%EB%A8%B8%EC%8A%A4%EC%86%94%EB%A3%A8%EC%85%98-%EC%82%AC%EC%9A%A9%EC%9E%90-%EC%8B%9D%EB%B3%84%EC%9E%90-%EC%97%B0%EB%8F%99-%EB%B0%8F-%EC%82%AC%EC%9A%A9-%ED%95%B4%EC%A7%80-%ED%94%84%EB%A1%9C%EC%84%B8%EC%8A%A4#%ED%8C%90%EB%A7%A4%EC%9E%90-%EC%BB%A4%EB%A8%B8%EC%8A%A4id-%EC%9D%B8%EC%A6%9D)
- [판매자 인증 JWE 구성요소 및 해석](https://github.com/commerce-api-naver/commerce-solutions/wiki/%5B%EA%B0%80%EC%9D%B4%EB%93%9C%5D-%ED%8C%90%EB%A7%A4%EC%9E%90-%EC%9D%B8%EC%A6%9D-JWE-%EA%B5%AC%EC%84%B1%EC%9A%94%EC%86%8C-%EB%B0%8F-%ED%95%B4%EC%84%9D)

## 세 번째 문의에 답변
이 공간의 Wiki 문서를 참고해주시기 바랍니다.
- https://github.com/commerce-api-naver/commerce-solutions/wiki
